dom/src/main/javacc/org/apache/james/mime4j/field/datetime/DateTimeParser.jj

```diff
@@ -225,7 +225,15 @@ int month() :
 String year() :
 {Token t;}
 {
-    t=<DIGITS> { return t.image; }
+    t=<DIGITS> 
+    {
+      String year = (String) t.image;
+      if (year.length() == 2) 

```

Comment:
Not sure that is enough ! Imagine "99" then you are returnning 2099 I would do "20" + year and then check it's not bigger than current year otherwise remove "100" but after all when the guy is not on the standard we can't really guess